lib/elixir/lib/enum.ex

```diff
@@ -1216,7 +1216,7 @@ defmodule Enum do
       "no bools!"
 
   """
-  @spec find_value(t, any, (element -> any)) :: any | default
+  @spec find_value(t, default, (element -> any)) :: any | default
```

Comment:
@whatyouhide what do you think about adding `mapped_element` like we have [here](https://hexdocs.pm/elixir/main/Enum.html#sort_by/3)?
Dialyzer might not get it, but might be helpful for the human reader (and might help with specs v2):

```suggestion
  @spec find_value(t, default, (element -> mapped_element)) :: mapped_element | default when mapped_element: term
```

Reply:
Sounds good to me. Maybe `found_element`. There is one important concept that `found_element` cannot be nil or false but we can't express it here (we can with set theoretic types though). Something such as `find_ok` would be better to express type wise but I even then `find_value` can be handy so I don't think it is necessarily worth introducing something new.

Reply:
Sure, we're fixing the the default here so I didn't comment outside of that, but yeah we can add `found_element`, which is still useful for docs IMO (even if it can' express the `nil`/`false` thing).

Reply:
@solar05 can you please apply this suggestion but using "found_value" instead?

Reply:
Sure, done 👌